clients/gtest/trmm_gtest.cpp

```diff
@@ -0,0 +1,75 @@
+/* ************************************************************************
+ * Copyright 2018-2019 Advanced Micro Devices, Inc.
+ *
+ * ************************************************************************ */
+
+#include "rocblas_data.hpp"
+#include "rocblas_datatype2string.hpp"
+#include "rocblas_test.hpp"
+#include "testing_trmm.hpp"
+#include "type_dispatch.hpp"
+#include <cctype>
+#include <cstring>
+#include <type_traits>
+
+namespace
+{
+    // By default, this test does not apply to any types.
+    // The unnamed second parameter is used for enable_if below.
+    template <typename, typename = void>
+    struct trmm_testing : rocblas_test_invalid
+    {
+    };
+
+    // When the condition in the second argument is satisfied, the type combination
+    // is valid. When the condition is false, this specialization does not apply.
+    template <typename T>
+    struct trmm_testing<
+        T,
+        typename std::enable_if<std::is_same<T, float>{} || std::is_same<T, double>{}>::type>
+    {
```

Comment:
The `trmm_testing` class needs to derive from `rocblas_test_valid` for the valid type combinations, and `rocblas_test_invalid` for the invalid type combinations.

Reply:
fixed in commit 560bdd2